drivers/adc/adc_shell.c

```diff
@@ -10,6 +10,26 @@
 #include <ctype.h>
 #include <sys/util.h>
 
+#if DT_HAS_COMPAT(atmel_sam_afec)
+#define DT_DRV_COMPAT atmel_sam_afec
+#elif DT_HAS_COMPAT(atmel_sam0_adc)
+#define DT_DRV_COMPAT atmel_sam0_adc
+#elif DT_HAS_COMPAT(microchip_xec_adc)
+#define DT_DRV_COMPAT microchip_xec_adc
+#elif DT_HAS_COMPAT(nordic_nrf_adc)
+#define DT_DRV_COMPAT nordic_nrf_adc
+#elif DT_HAS_COMPAT(nordic_nrf_saadc)
+#define DT_DRV_COMPAT nordic_nrf_saadc
+#elif DT_HAS_COMPAT(nxp_kinetis_adc12)
+#define DT_DRV_COMPAT nxp_kinetis_adc12
+#elif DT_HAS_COMPAT(nxp_kinetis_adc16)
+#define DT_DRV_COMPAT nxp_kinetis_adc16
+#elif DT_HAS_COMPAT(st_stm32_adc)
+#define DT_DRV_COMPAT st_stm32_adc
+#else
+#error No know devicetree compatible match for ADC shell
+#endif
+
```

Comment:
This will prevent the ADC shell from working with multiple ADC drivers at once (e.g. SoC ADC and external, SPI ADC).

Reply:
That limitation exist today.  There is no support for ADC shell on non-SoC ADC currently.  I'd ask we look at doing that as a follow up to this change.

Reply:
Fair enough.